node/src/proxy_server/mod.rs

```diff
@@ -3635,13 +3667,52 @@ mod tests {
             scheduled_msg,
             &MessageScheduler {
                 scheduled_msg: StreamKeyPurge { stream_key },
-                delay: Duration::from_secs(5)
+                delay: STREAM_KEY_PURGE_DELAY
             }
-        )
+        );
+        TestLogHandler::new()
+            .exists_log_containing(&format!(
+                "DEBUG: {test_name}: Straggling packet received for a stream key {stream_key} with a delay of"
+            ));
+    }
+
+    #[test]
+    #[should_panic(expected = "time calculation error")]
+    fn log_straggling_packet_panics_if_timestamp_is_wrong() {
+        let subject = ProxyServer::new(
+            main_cryptde(),
+            alias_cryptde(),
+            true,
+            Some(STANDARD_CONSUMING_WALLET_BALANCE),
+            false,
+        );
+        let stream_key = StreamKey::make_meaningless_stream_key();
+        let timestamp = SystemTime::now()
+            .checked_add(Duration::from_secs(10))
+            .unwrap();
+        let _ = subject.log_straggling_packet(&stream_key, &timestamp);
     }
 
     #[test]
     fn handle_client_response_payload_purges_stream_keys_for_terminal_response() {
```

Comment:
You know this already, but just to mark this location: this is about receiving `last_data = true` rather than sending it, so the delayed-purge logic shouldn't come into play here.